frontend/src/plugins/global_mixin.js

```diff
@@ -33,13 +51,219 @@ var GlobalMixin = {
         },
     },
     methods: {
+        async clear() {
+            await this.$store.dispatch("images/clear")
+        },
+        async UploadExists(filename) {
+            const uploads_exists = this.uploads.find(
+                (i) => i.filename === filename
+            )
+            const inprogress_exists = this.inprogress.find(
+                (i) => i.filename === filename
+            )
+            let exists = []
+            if (uploads_exists) {
+                exists.push("uploads")
+            }
+            if (inprogress_exists) {
+                exists.push("inprogress")
+            }
+            if (exists.length > 0) {
+                this.$info(
+                    `File ${filename} already exists in ${exists.join(", ")}`
+                )
+                return true
+            } else {
+                return false
+            }
+        },
         async get_ref(ref_id) {
             while (!this.$refs[ref_id]) {
                 await new Promise((resolve) => setTimeout(resolve, 100))
             }
             return this.$refs[ref_id]
         },
+        async doremove(sub) {
+            console.log("remove", sub)
+            await this.$store.dispatch("images/remove", sub.hash)
+        },
+        async doupload(files) {
+            ;[...files].forEach(async (file) => {
+                const reader = new FileReader()
```

Comment:
note for myself, there is a function I made for this that has some error handling.